halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/kubernetes/KubernetesAccountValidator.java

```diff
@@ -0,0 +1,156 @@
+import com.netflix.spinnaker.halyard.config.model.v1.node.Node;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.ProblemSetBuilder;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.dockerRegistry.DockerRegistryProvider;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.kubernetes.DockerRegistryReference;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.kubernetes.KubernetesAccount;
+import io.fabric8.kubernetes.api.model.NamedContext;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.internal.KubeConfigUtils;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.netflix.spinnaker.halyard.config.model.v1.problem.Problem.Severity.ERROR;
+import static com.netflix.spinnaker.halyard.config.model.v1.problem.Problem.Severity.WARNING;
+
+@Component
+public class KubernetesAccountValidator extends Validator<KubernetesAccount> {
+  @Override
+  public void validate(ProblemSetBuilder psBuilder, KubernetesAccount account) {
+    DeploymentConfiguration deploymentConfiguration;
+
+    // TODO(lwander) this is still a little messy - I should use the references to get the necessary docker account
+    Node parent = account.getParent();
+    while (!(parent instanceof DeploymentConfiguration)) {
+      // Note this will crash in the above check if the halconfig representation is corrupted
+      // (that's ok, because it indicates a more serious error than we want to validate).
+      parent = parent.getParent();
+    }
+    deploymentConfiguration = (DeploymentConfiguration) parent;
+
+    validateDockerRegistries(psBuilder, account, deploymentConfiguration);
+    validateKubeconfig(psBuilder, account);
+  }
+
+  private void validateKubeconfig(ProblemSetBuilder psBuilder, KubernetesAccount account) {
+    io.fabric8.kubernetes.api.model.Config kubeconfig;
+    String context = account.getContext();
+    String kubeconfigFile = account.getKubeconfigFile();
+    String cluster = account.getCluster();
+    String user = account.getUser() ;
+    List<String> namespaces = account.getNamespaces();
+
+    // This indicates if a first pass at the config looks OK. If we don't see any serious problems, we'll do one last check
+    // against the requested kubernetes cluster to ensure that we can run spinnaker.
+    boolean smoketest = true;
+
+    try {
+      File kubeconfigFileOpen = new File(kubeconfigFile);
+      kubeconfig = KubeConfigUtils.parseConfig(kubeconfigFileOpen);
+    } catch (IOException e) {
+      psBuilder.addProblem(ERROR, e.getMessage());
```

Comment:
If the problem were there is no kubeconfig file, would you want to instruct how to create one (or point to instructions)?